presto-common/src/main/java/com/facebook/presto/common/Page.java

```diff
@@ -456,6 +456,24 @@ private long updateRetainedSize()
         return retainedSizeInBytes.longValue();
     }
 
+    /**
+     * Returns a new page with the same columns as the original page except for the one column replaced.
+     *
+     * @param channelIndex the column to replace
+     * @param column the replacement column
+     * @return a new page with the replacement column substituted for the old column
+     */
+    public Page replaceColumn(int channelIndex, Block column)
+    {
+        if (column.getPositionCount() != positionCount) {
+            throw new IllegalArgumentException("New column does not have same number of rows as old column");
```

Comment:
Replace: same number of rows as old column -> same number of rows as the page.

Technically current page implementation allows to have blocks with different number of rows, but I think it's ok to enforce this constraint.

Reply:
appendColumn and prependColumn do check for this though interestingly, as you point out, the constructor does not.